engine/api/src/main/scala/pl/touk/nussknacker/engine/api/ConfigLoader.scala

```diff
@@ -0,0 +1,24 @@
+package pl.touk.nussknacker.engine.api
+
+import com.typesafe.config.Config
+
+/**
+  * Loads base process config. Resolved config should be available when a process is run
+  * (resolved config is provided to the process by a process manager).
+  *
+  * Config is resolved once when loading model data. Implement this trait when you want to add dynamic configuration
+  * that is fetched from external source, for example OpenAPI specification.
+  *
+  * This trait is optional to implement.
+  */
+trait ConfigLoader {
```

Comment:
1. I see a problem that it is not clear what exactly happen during config loading. Part of logic is in `ModelConfigToLoad.load`, part will be here. WDYT about extracting new object e.g. `DefaultModelConfigLoader` which will replace  `ConfigLoader.identity` and will do what is done in `ModelConfigToLoad.load`? If smb want to reuse it in theirs own implementation, he/she can just invoke it in any place.
2. Also IMO we should rename it to `ModelConfigLoader` because we have many configs in our projects and this name is confusing